### PR TITLE
Post request

### LINKED ISSUE #7: Ticket 2b - POST REQUEST (Developer only)
| name  | about | title  | labels | assignees  |
| ------------- | ------------- | ------------- | ------------- | ------------- |
| "👥 User story"  | Create a new user story  | LINK API  | POST REQUEST | Saghar  | 

## Story: POST REQUEST (developer only) From Link API

- *As a ….developer,

- **I want to….. link the GitHub API to the backend

- *So That ….. I can add data from the different GitHub repos to the server.

## Acceptance Criteria

1.GIVEN ... I want a user to add the GitHub PR contributions of any trainee
- WHEN ...I click on the trainees GitHub Link
- THEN ...I should be able to add trainee’s GitHub PR’s to the server

2. Given .............
   - When .............
   - Then .............

## Implementation Notes

1.
2.

## Checklist:

- [ ] I have defined the user story
- [ ] I have defined some acceptance criteria
- [ ] I have included any implementation notes
- [ ] I have included all relevant links

### PR DESCRIPTION
**Pull request by SagharHosseinmardi**
#7 -POST REQUEST-SAGHAR HOSSEINMARDI

## Description

UPDATED THE POST REQUEST IN SERVER SIDE

## Related to #7 

Fixes #7 

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have carefully reviewed my own code
- [ ] I have commented my code
- [ ] I have updated any documentation

### CODE REVIEW

File: client/src/components/Form.js [hunk 1]
```diff
@@ -0,0 +1,159 @@
+import React, { useState } from "react";
```
Comment (AndrianaOS) [inline client/src/components/Form.js]: You need to remove this file from this branch so its server side code only 
Form already has its own branch

File: server/server.js [hunk 2]
```diff
@@ -0,0 +1,66 @@
+import express from "express";
```
Comment (AndrianaOS) [inline server/server.js]: This file also needs to be removed as the working code is inn index.js already

File: server/index.js [hunk 3]
```diff
@@ -1,24 +1,100 @@
...
+        INSERT INTO fp_members (team_id, member_name, role, github_username)
+        VALUES ($1, $2, $3, $4)
+      `;
+
+      for (const member of members) {
+        const { memberName, role, githubUsername } = member;
+        const memberValues = [teamId, memberName, role, githubUsername];
+        await dataBase.query(memberInsertQuery, memberValues);
+      }
+      res.json({ success: true });
```
Comment (AndrianaOS) [inline server/index.js]: return a status then maybe a message. Something like:
`res.status(200).json({message: "Team information submitted"})`

File: server/package.json [hunk 4]
```diff
@@ -2,10 +2,10 @@
   "name": "server",
   "version": "1.0.0",
   "description": "",
-  "main": "index.js",
+  "main": "server.js",
   "type": "module",
   "scripts": {
-    "start": "nodemon index.js",
+    "start": "nodemon server.js",
```
Comment (AndrianaOS) [inline server/package.json]: You need to change the start script back to "nodemon index.js"

File: server/package.json [hunk 5]
```diff
@@ -2,10 +2,10 @@
   "name": "server",
   "version": "1.0.0",
   "description": "",
-  "main": "index.js",
+  "main": "server.js",
```
Comment (AndrianaOS) [inline server/package.json]: This also needs to change back to index.js

**Review by AndrianaOS — CHANGES_REQUESTED** [review]

**Review by AndrianaOS — CHANGES_REQUESTED** [review]
Just a few changes that will affect the code running if merged. Also removal of files that belong in another branch

File: client/src/components/Form.js (hunk 1 above)
Comment (SagharHosseinmardi) [reply client/src/components/Form.js]: this part is done.

File: server/server.js (hunk 2 above)
Comment (SagharHosseinmardi) [reply server/server.js]: this part is done.

File: server/index.js (hunk 3 above)
Comment (SagharHosseinmardi) [reply server/index.js]: This part is done.

File: server/package.json (hunk 4 above)
Comment (SagharHosseinmardi) [reply server/package.json]: This part is done.

File: server/package.json (hunk 5 above)
Comment (SagharHosseinmardi) [reply server/package.json]: This part is done.

**Comment by SagharHosseinmardi** [comment]
i have done all the changes.

**Review by AndrianaOS — APPROVED** [review]
Ready to be merged

**Pull request merged**